### PR TITLE
Facility Data Module upgradation for OpenMRS 2.x (All changes in one commit)

### PR DESCRIPTION
**Pull request by mahmed14**
@dkayiwa  , i have squashed all changes into one commit please look into this pull request.

### CODE REVIEW

**Comment by dkayiwa** [comment]
Can you also include a link to the JIRA ticket as advised at https://wiki.openmrs.org/display/docs/Pull+Request+Tips

File: api/.classpath [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" output="target/classes" path="src/main/java">
+		<attributes>
+			<attribute name="optional" value="true"/>
+			<attribute name="maven.pomderived" value="true"/>
```
Comment (dkayiwa) [inline api/.classpath]: We do not commit IDE files like .classpath

File: api/.project [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.springframework.ide.eclipse.core.springbuilder</name>
```
Comment (dkayiwa) [inline api/.project]: Same as above

File: api/.settings/org.eclipse.jdt.core.prefs [hunk 3]
```diff
@@ -0,0 +1,5 @@
+eclipse.preferences.version=1
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.6
```
Comment (dkayiwa) [inline api/.settings/org.eclipse.jdt.core.prefs]: Same as above

File: api/.settings/org.eclipse.m2e.core.prefs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+activeProfiles=
+eclipse.preferences.version=1
```
Comment (dkayiwa) [inline api/.settings/org.eclipse.m2e.core.prefs]: Same as above

File: api/src/main/java/org/openmrs/module/facilitydata/model/CodedFacilityDataQuestionType.java [hunk 5]
```diff
@@ -13,17 +13,24 @@
  */
 package org.openmrs.module.facilitydata.model;
 
+
+import javax.persistence.*;
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/facilitydata/model/CodedFacilityDataQuestionType.java]: We do not encourage wild card imports https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotusewildcardimports

File: api/src/main/java/org/openmrs/module/facilitydata/model/CodedFacilityDataQuestionType.java [hunk 6]
```diff
@@ -85,4 +90,9 @@ public FacilityDataCodedOption getOptionById(Integer id) {
 		}
 		return ret;
 	}
+
+	public void addFacilityDataCodedOption(FacilityDataCodedOption child) {
+		//child.setQuestionType(this);
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/facilitydata/model/CodedFacilityDataQuestionType.java]: Commented out code is better removed https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotcommentoutcode

**Comment by mahmed14** [comment]
@dkayiwa  all these changes include support for OpenMRS 2.x and other functionalities also . Like textQuestions , upgrade to liquibase, Document type question (XML,JSON) and Blob questionatype and some rest implementations also

**Comment by dkayiwa** [comment]
It is not good to have such a big pull request that does so many things. My recommendation is that you separate these in different tickets and hence pull requests. For instance, a ticket and corresponding pull request about supporting platform 2.x should have only those changes required for the module to run on platform 2.x and no more. Then another ticket and pull request for another logically related work. And more.

**Comment by mahmed14** [comment]
@dkayiwa ,  this would require alot of rework at my end and currently i am busy with alot of others things. is there any other way. i did talk with mike seaton on talk [https://talk.openmrs.org/t/facility-data-module-management-and-code-review/15887/11](url) . is there any other way which might require less time.

File: api/src/main/java/org/openmrs/module/facilitydata/model/CodedFacilityDataQuestionType.java (hunk 6 above)
Comment (mahmed14) [reply api/src/main/java/org/openmrs/module/facilitydata/model/CodedFacilityDataQuestionType.java]: ok

**Comment by mahmed14** [comment]
@dkayiwa ,

any suggestion ??

**Comment by dkayiwa** [comment]
I do not know the internals of this module for i have never used it. But am an expert in making modules run on platform 2.x and was anxiously ready to review, advise, and merge that part. But since you mixed it with other things that have nothing to do with just making the module run on platform 2.x, then am not in the best position to review and merge these combined changes.

**Comment by mahmed14** [comment]
@dkayiwa , ok then should i create a separate pull request for only OpenMRS 2.x ? the code will only include support for 2.x and then send all these changes in one pull request ?

**Comment by dkayiwa** [comment]
If you do so, then you will have equipped me to review, advise, and merge it. 😊

**Comment by mahmed14** [comment]
ok, just give me some time as i have gone beyond that . so i have to do little rework again and currently i have others things on my plate too.

**Comment by mahmed14** [comment]
@dkayiwa  you can also give some advance tips regarding upgrade . :-)

**Comment by dkayiwa** [comment]
Do they pull request and i will advise while reviewing.
Just in case you find these helpful:
https://wiki.openmrs.org/display/docs/Supporting+different+OpenMRS+versions
https://wiki.openmrs.org/display/docs/Supporting+Platform+2.0+and+below

**Comment by dkayiwa** [comment]
@mahmed14 are you still working on this?

**Comment by dkayiwa** [comment]
@mahmed14 ping me to reopen this when you resume working on it. 😊

**Comment by mahmed14** [comment]
@dkayiwa  can we have a skype call? need to discuss some points regarding facility data module

**Comment by mahmed14** [comment]
can we have it on skype??

**Comment by dkayiwa** [comment]
Skype requires me to first pay the daily taxes. Do you have any particular problem with? http://uberconference.com/openmrs

**Comment by mahmed14** [comment]
no , i don't any issue with it. when should we have the call?

**Comment by dkayiwa** [comment]
Whenever you are ready.

**Comment by mahmed14** [comment]
can we have call  today or tomorrow??

**Comment by dkayiwa** [comment]
As for me, am available any time you want.

**Comment by mahmed14** [comment]
okay , lets have it after 30mins . is it okay for you ??? @dkayiwa

**Comment by dkayiwa** [comment]
Cool.

**Comment by mahmed14** [comment]
@dkayiwa i am on the call

**Pull request closed without merging**